src/BBitsEmoji.sol

```diff
@@ -0,0 +1,181 @@
+    /// @dev    raffleId => totalEntries
+    mapping(uint256 => uint256) public totalEntries;
+
+    /// @dev Begins paused to allow owner to add art.
+    constructor(address _owner, address _burner, IBBitsCheckIn _checkin) ERC1155("") Ownable(_owner) {
+        burner = Burner(_burner);
+        checkIn = _checkin;
+        burnPercentage = 4000;
+        mintDuration = 1 days;
+        mintPrice = 0.0005 ether;
+        totalEntries[currentDay] = 1;
+        _pause();
+    }
+
+    receive() external payable {}
+
+    /// @notice This function provides the core functionality for this contract, including minting, creating
+    ///         raffles, settling raffles, and generating new daily art.
+    function mint() external payable nonReentrant whenNotPaused {
+        if (willMintSettleRaffle()) {
+            _startNewMint();
+            _mintEntry();
+        } else {
+            if (msg.value < userMintPrice(msg.sender)) revert MustPayMintPrice();
+            _mintEntry();
+        }
+    }
+
+    /// OWNER ///
+
+    function setPaused(bool _setPaused) external onlyOwner {
+        _setPaused ? _pause() : _unpause();
+    }
+
+    function setMintPrice(uint256 _mintPrice) external onlyOwner {
+        mintPrice = _mintPrice;
+    }
+
+    /// VIEW ///
+
+    /// @notice This view function returns the art for any given token Id.
+    /// @param  _tokenId The token Id of the NFT.
+    function uri(uint256 _tokenId) public view override returns (string memory) {
+        return _draw(_tokenId);
+    }
+
+    /// @notice This view function returns the current amount of ETH that will be awwarded to the daily
+    ///         raffle winner.
+    function raffleAmount() public view returns (uint256) {
+        return address(this).balance - burnAmount();
+    }
+
+    /// @notice This view function returns the current amount of ETH that will be used to buy and burn
+    ///         the BBITS token.
+    function burnAmount() public view returns (uint256) {
+        return (burnPercentage * address(this).balance) / 10_000;
+    }
+
+    /// @notice This view function returns the current price to mint an NFT for any given user. Users can
+    ///         decrease their mint price by up to 90% by gaining a streak in the associated BBits CheckIn
+    ///         contract: 0xE842537260634175891925F058498F9099C102eB.
+    /// @param  _user The user paying to mint a new NFT.
+    function userMintPrice(address _user) public view returns (uint256) {
+        (, uint16 streak,) = checkIn.checkIns(_user);
+        if (streak > 90) streak = 90;
+        return mintPrice - ((mintPrice * streak) / 100);
+    }
+
+    /// @notice This view function returns the entry struct that corresponds to the day and index pair.
+    /// @param  _day The raffle day (token Id).
+    /// @param  _index The user index in the raffleInfo array of raffle entries.
+    function userEntry(uint256 _day, uint256 _index) public view returns (Entry memory) {
```

Comment:
Can we rename this method to `useEntryByIdex` and add another method to return entry by address `userEntry(uint256 _day, address _user)` ? 

This way I will be able to display user's info in the UI without the need to loop through all the entries.